fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/ReduceCastRule.java

```diff
@@ -110,17 +110,18 @@ public boolean checkCastTypeReduceAble(Type parent, Type child, Type grandChild)
             return false;
         }
 
-        // decreasing cast cannot be reduced since be use operator not equal
-        // to perform cast from integral to boolean
-        // E.g. cast(cast(10000000000000 as int) as boolean)
-        if (parent.isBoolean() && childSlotSize < grandChildSlotSize) {
+        if (!(parent.isNumericType() || parent.isStringType()) ||
```

Comment:
Can other types benefit from it?

Reply:
Maybe, but it's hard to check each type, so only support Number/String/Boolean type